filament/include/filament/Material.h

```diff
@@ -211,6 +211,9 @@ class UTILS_PUBLIC Material : public FilamentAPI {
     //! Indicates whether a parameter of the given name exists on this material.
     bool hasParameter(const char* name) const noexcept;
 
+    //! Indicates whether an existing parameter is a sampler or not.
+    bool hasSampler(const char* name) const noexcept;
```

Comment:
Wouldn't it make more sense for this to be called `isSampler`? `hasParameter` above will tell you whether a parameter exists on the material. Here you want to check the nature of said parameter. Actually a better API would be `getParameterType(const char*)`, so it would work with vectors, matrices, etc.

Reply:
I agree, I will rename it to `isSampler()`. 

I'm not sure about `getParameterType()` . What would this function return as the type is stored as a union:

```c++
    struct ParameterInfo {
        //! Name of the parameter.
        const char* name;
        //! Whether the parameter is a sampler (texture).
        bool isSampler;
        union {
            //! Type of the parameter if the parameter is not a sampler.
            ParameterType type;
            //! Type of the parameter if the parameter is a sampler.
            SamplerType samplerType;
        };
        //! Size of the parameter when the parameter is an array.
        uint32_t count;
        //! Requested precision of the parameter.
        Precision precision;
    };
```